src/FormNetworks/bilinearformnetwork.jl

```diff
@@ -0,0 +1,62 @@
+default_bra_vertex_map(v) = (v, "bra")
+default_ket_vertex_map(v) = (v, "ket")
+default_operator_vertex_map(v) = (v, "operator")
+default_operator_constructor(s::IndsNetwork) = delta_network(s)
+
+struct BilinearFormNetwork{V,KetMap,BraMap,OperatorMap} <: AbstractFormNetwork{V}
+  tn::AbstractITensorNetwork{V}
+  bra_vertex_map::BraMap
+  ket_vertex_map::KetMap
+  operator_vertex_map::OperatorMap
+end
+
+function BilinearFormNetwork(
+  operator::AbstractITensorNetwork,
+  bra::AbstractITensorNetwork,
+  ket::AbstractITensorNetwork;
+  bra_vertex_map=default_bra_vertex_map,
+  ket_vertex_map=default_ket_vertex_map,
+  operator_vertex_map=default_operator_vertex_map,
+)
+  @assert Set(externalinds(operator)) ==
+    union(Set(externalinds(ket)), Set(externalinds(bra)))
+  @assert isempty(findall(in(internalinds(bra)), internalinds(ket)))
+  @assert isempty(findall(in(internalinds(bra)), internalinds(operator)))
+  @assert isempty(findall(in(internalinds(ket)), internalinds(operator)))
```

Comment:
I think we can just remove these checks for now to keep the code simpler, and see in practice what kinds of mistakes are made. My guess is that if there are index issues they can be caught somewhere else, lower down (say in the `ITensorNetwork` constructor, or somewhere in the Krylov solver).